Firestore/core/src/firebase/firestore/util/CMakeLists.txt

```diff
@@ -72,41 +62,49 @@ cc_select(
 )
 
 
-## log
+## base
 
 cc_library(
-  firebase_firestore_util_log_stdio
+  firebase_firestore_util_base_stdio
   SOURCES
     hard_assert_stdio.cc
     hard_assert.h
     log.h
     log_stdio.cc
+    string_format.cc
+    string_format.h
+    string_win.cc
+    string_win.h
   DEPENDS
     firebase_firestore_util_base
     absl_base
+    absl_strings
   EXCLUDE_FROM_ALL
 )
 
 cc_library(
-  firebase_firestore_util_log_apple
+  firebase_firestore_util_base_apple
   SOURCES
     hard_assert.h
     hard_assert_apple.mm
     log.h
     log_apple.mm
     string_apple.h
+    string_format.cc
+    string_format.h
   DEPENDS
     FirebaseCore
     GoogleUtilities
+    absl_base
     absl_strings
     firebase_firestore_util_base
   EXCLUDE_FROM_ALL
 )
 
 cc_select(
-  firebase_firestore_util_log
-  APPLE   firebase_firestore_util_log_apple
-  DEFAULT firebase_firestore_util_log_stdio
+  firebase_firestore_util_base
+  APPLE   firebase_firestore_util_base_apple
+  DEFAULT firebase_firestore_util_base_stdio
```

Comment:
IIUC, cc_select makes _base be implemented by either _base_apple or _base_stdio. Which would seem to make the depends statements on lines 79,100 redundant.

Reply:
That's actually a bug: it makes the dependencies circular. Removed.